databricks_cli/configure/config.py

```diff
@@ -47,11 +49,19 @@ def decorator(*args, **kwargs):
     return decorator
 
 
-def profile_option(f):
+def profile_option_injected(f):
     return click.option('--profile', required=False, default=DEFAULT_SECTION,
                         help='CLI connection profile to use.')(f)
 
 
+def profile_option(f):
+    def callback(ctx, param, value): #  NOQA
+        context_object = ctx.ensure_object(ContextObject)
+        context_object.set_profile(value)
+    return click.option('--profile', required=False, default=DEFAULT_SECTION, callback=callback,
```

Comment:
Should mention which one takes over if there are multiple `--profile` specified.